src/abTestingInfra/ABTestingContext/index.tsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import { AB_TEST_PAYLOAD_PREFIX } from '../utils';
+import { useRouter } from 'next/router';
+import { ABTestingPayload } from '../types';
+
+const defaultABTestingContext: ABTestingPayload = {};
+const ABTestingContext = React.createContext(defaultABTestingContext);
+
+function getABTestContext(payload: string): ABTestingPayload {
+  const searchParams = new URLSearchParams(payload);
+
+  const result: ABTestingPayload = {};
+
+  // can't call reduce on searchParams, that's why using forEach
+  searchParams.forEach((value, key) => {
+    result[key] = value;
+  });
+
+  return result;
+}
+
+function getABTestContextFromPath(path: string): ABTestingPayload {
+  const idx = path.indexOf(AB_TEST_PAYLOAD_PREFIX);
+
+  if (idx === -1) {
+    return {};
+  }
+
+  // removing possible trailing slash
+  const normalizedPath = path.replace(/\/$/, '');
+
+  // we should exclude possible query string from payload
+  const queryIdx = path.indexOf('?');
+  let endIdx = normalizedPath.length;
+
+  if (queryIdx !== -1) {
+    endIdx = queryIdx;
+  }
```

Comment:
1/ Same as above, re naming
2/ Should this operation be reversed with the above? Consider URLs with a trailing slash before the query string, e.g. `https://google.com/search/?q=foo`
3/ What is the value of doing this as a slice, vs. `foo.replace(/\/?(\?.*)?/, '')`

Reply:
Replaced with regex